src/app/service/product/product.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProductService {
+
+  url = "http://localhost:9092/api/products";
+
+  constructor(private httpClient: HttpClient) { }
+
+  public getAllProducts() {
+    console.log(this.url)
+    return this.httpClient.get(this.url);
+  }
+
+  public getProductById(userId: string, sellerDetail: boolean = false) {
+    return this.httpClient.get(this.url + "/" + userId + "?" + "sellerDetail=" + sellerDetail);
```

Comment:
you should use `${this.url}/${userId}` like this in typescript

Reply:
Done.